doc/api/https.md

```diff
@@ -251,6 +252,101 @@ const req = https.request(options, (res) => {
 });
 ```
 
+Example pinning on certificate fingerprint, or the public key (similar to `pin-sha256`):
+
+```js
+const tls = require('tls');
+const https = require('https');
+const crypto = require('crypto');
+
+const options = {
+  checkServerIdentity: function(host, cert) {
+    /* Make sure the certificate is issued to the host we are connected to */
+    const err = tls.checkServerIdentity(host, cert);
+    if (err) {
+      return err;
+    }
+
+    /* Pin the public key, similar to HPKP pin-sha25 pinning */
+    const pubkey256 = 'pL1+qb9HTMRZJmuC/bB/ZI9d302BYrrqiVuRyW+DGrU=';
+    hash = crypto.createHash('sha256');
+    hash.update(cert.pubkey);
+    if (hash.digest('base64') !== pubkey256) {
```

Comment:
For legibility/succinctness/DRY, maybe put this in a separate function:
```js
function sha256(s) {
  return crypto.createHash('sha256').update(s).digest('base64');
}
```
And then you can write:
```js
if (pubkey256 !== sha256(cert.pubkey)) {
```
And below:
```js
console.log('\tPublic key ping-sha256:', sha256(cert.pubkey));  // also note the , instead of +
```